Source/Scene/CesiumIon.js

```diff
@@ -0,0 +1,267 @@
+    }
+
+    CesiumIonResource.create = function(endpoint, endpointResource) {
+        var options = {
+            url: endpoint.url,
+            retryCallback: createRetryCallback(endpoint, endpointResource),
+            retryAttempts: 1
+        };
+
+        if (defined(endpoint.accessToken)) {
+            options.queryParameters = { access_token: endpoint.accessToken };
+        }
+
+        return new CesiumIonResource(options, endpoint, endpointResource);
+    };
+
+    if (defined(Object.create)) {
+        CesiumIonResource.prototype = Object.create(Resource.prototype);
+        CesiumIonResource.prototype.constructor = CesiumIonResource;
+    }
+
+    CesiumIonResource.prototype.clone = function(result) {
+        if (!defined(result)) {
+            result = new CesiumIonResource({
+                url: this._url
+            });
+        }
+        result = Resource.prototype.clone.call(this, result);
+        result.ionData = this.ionData;
+        return result;
+    };
+
+    function createRetryCallback(endpoint, endpointResource) {
+        return function(that, error) {
+            // We only want to retry in the case of invalid credentials (401) or image
+            // requests(since Image failures can not provide a status code)
+            if (!defined(error) || (error.statusCode !== 401 && !(error.target instanceof Image))) {
+                return when.resolve(false);
+            }
+
+            var ionData = that.ionData;
+
+            // We use a shared pending promise for all derived assets, since they share
+            // a common access_token.  If we're already requesting a new token for this
+            // asset, we wait on the same promise.
+            var pendingPromise = ionData._pendingPromise;
+            if (!defined(pendingPromise)) {
+                pendingPromise = CesiumIon._loadJson(endpointResource)
+                    .then(function(endpoint) {
+                        //Set the token for root resource that this (and other) resources were derived
+                        ionData._root.queryParameters.access_token = endpoint.accessToken;
+                        return endpoint.accessToken;
+                    })
+                    .always(function(accessToken) {
+                        // Pass or fail, we're done with this promise, the next failure should use a new one.
+                        ionData._pendingPromise = undefined;
+
+                        // We need this return because our old busted version of when
+                        // doesn't conform to spec of returning the result of the above `then`.
+                        return accessToken;
+                    });
+                ionData._pendingPromise = pendingPromise;
+            }
+
+            return pendingPromise.then(function(accessToken) {
+                // Set the new token for this resource
+                that.queryParameters.access_token = accessToken;
+                return true;
+            });
```

Comment:
`.otherwise`?

Reply:
We are returning this promise for the code handling the retry to consume, so it's their job to handle the otherwise (which currently happens in loadWithXhr and loadImage I believe).